eZ/Publish/Core/REST/Server/Output/ValueObjectVisitor/RestExecutedView.php

```diff
@@ -108,16 +113,26 @@ public function visit(Visitor $visitor, Generator $generator, $data)
 
             $generator->startObjectElement('value');
 
-            /** @var \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo */
-            $contentInfo = $searchHit->valueObject->contentInfo;
-            $restContent = new RestContentValue(
-                $contentInfo,
-                $this->locationService->loadLocation($contentInfo->mainLocationId),
-                $searchHit->valueObject,
-                $this->contentTypeService->loadContentType($contentInfo->contentTypeId),
-                $this->contentService->loadRelations($searchHit->valueObject->getVersionInfo())
-            );
-            $visitor->visitValueObject($restContent);
+            // @todo Refactor
+            if ($searchHit->valueObject instanceof Content) {
+                /** @var \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo */
+                $contentInfo = $searchHit->valueObject->contentInfo;
+                $valueObject = new RestContentValue(
+                    $contentInfo,
+                    $this->locationService->loadLocation($contentInfo->mainLocationId),
+                    $searchHit->valueObject,
+                    $this->contentTypeService->loadContentType($contentInfo->contentTypeId),
+                    $this->contentService->loadRelations($searchHit->valueObject->getVersionInfo())
+                );
+            } elseif ($searchHit->valueObject instanceof LocationValueObject) {
+                $valueObject = new RestLocationValueObject($searchHit->valueObject, 0);
+            } elseif ($searchHit->valueObject instanceof ContentInfo) {
+                $valueObject = new RestContent($searchHit->valueObject);
```

Comment:
RestContent _is_ a Value object. It's one of those decorated ones we use for Rest return values.